src/servers/ZoneServer2016/entities/plantingdiameter.ts

```diff
@@ -46,4 +52,49 @@ export class PlantingDiameter extends TemporaryEntity {
     }
     return server.deleteEntity(this.characterId, server._temporaryObjects);
   }
+
+  OnPlayerSelect(server: ZoneServer2016) {
+    for (const clients in server._clients) {
+      const client = server._clients[clients];
+
+      if (this.canUndoPlacement(server, client)) {
+        this.destroy(server);
+        client.character.lootItem(
+          server,
+          server.generateItem(Items.GROUND_TILLER)
+        );
+        return;
+      }
+    }
+  }
+
+  OnMeleeHit(server: ZoneServer2016, damageInfo: DamageInfo) {
+    const client = server.getClientByCharId(damageInfo.entity),
+      weapon = client?.character.getEquippedWeapon();
+    if (!client || !weapon) return;
+    if (weapon.itemDefinitionId !== Items.WEAPON_HAMMER_DEMOLITION) return;
+    for (const plant of Object.values(this.seedSlots)) {
+      plant.destroy(server);
+    }
+    server.deleteEntity(this.characterId, server._temporaryObjects);
+  }
+
+  canUndoPlacement(server: ZoneServer2016, client: ZoneClient2016) {
+    const weapon = client.character.getEquippedWeapon();
+    if (!weapon) return false;
+
+    for (const a in server._constructionFoundations) {
```

Comment:
This doesn't feel right this should undo the placement of the plantingdiameter entity not iterate on all foundations, doesn't it destroy all entities that match ?